homeassistant/components/tado/__init__.py

```diff
@@ -172,13 +184,45 @@ def setup(self):
         self.home_id = tado_home["id"]
         self.home_name = tado_home["name"]
 
+    def get_me(self):
+        """Return the Tado 'me' data."""
+        return self.tado.getMe()
+
+    def get_mobile_devices(self):
+        """Return the Tado mobile devices."""
+        return self.tado.getMobileDevices()
+
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Update the registered zones."""
         self.update_devices()
         self.update_zones()
         self.update_home()
 
+    def update_mobile_devices(self) -> None:
+        """Update the mobile devices."""
+        try:
+            mobile_devices = self.get_mobile_devices()
+        except RuntimeError:
+            _LOGGER.error("Unable to connect to Tado while updating mobile devices")
+            return
+
+        for mobile_device in mobile_devices:
+            self.data["mobile_device"][mobile_device["id"]] = mobile_device
+
+            _LOGGER.debug(
+                "Dispatching update to %s mobile device %s: %s",
+                self.home_id,
+                mobile_device["id"],
+                mobile_device,
+            )
+            dispatcher_send(
+                self.hass,
+                SIGNAL_TADO_MOBILE_DEVICE_UPDATE_RECEIVED.format(
+                    "mobile_device", mobile_device["id"]
```

Comment:
The problem with sending a signal specific for a single device is that we can't listen for new devices. I suggest sending a general signal when mobile devices are updated.